scripts/dockerbuild.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+set -e
+
+SOURCE="${BASH_SOURCE[0]}"
+while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
+  DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+  SOURCE="$(readlink "$SOURCE")"
+  [[ "$SOURCE" != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+done
+DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+
+cd "$DIR/.."
+
+INTERACTIVE="-i"
+
+while [[ $# > 0 ]]; do
+    key=$1
+
+    case $key in
+        --non-interactive)
+            INTERACTIVE=
+            ;;
+        -t|--tag)
+            DOCKER_TAG=$2
+            shift
+            ;;
+        -d|--dockerfile)
+            DOCKERFILE=$2
+            shift
+            ;;
+        -h|-?|--help)
+            echo "Usage: $0 [-d|--dockerfile <Dockerfile>] [-t|--tag <Tag>] <Command>"
+            echo ""
+            echo "Options:"
+            echo "  <Dockerfile>    The path to the folder that contains a Dockerfile to use to create the build container"
+            echo "  <Tag>           The name of docker image tag"
+            exit 0
+            ;;
+        *)
+            break # the first non-switch we get ends parsing
+            ;;
+    esac
+
+    shift
+done
+
+# Make container names CI-specific if we're running in CI
+#  Jenkins
+[ ! -z "$BUILD_TAG" ] && DOTNET_BUILD_CONTAINER_NAME="${BUILD_TAG}-${DOCKER_TAG}"
+#  VSO
+[ ! -z "$BUILD_BUILDID" ] && DOTNET_BUILD_CONTAINER_NAME="${BUILD_BUILDID}-${BUILD_BUILDNUMBER}-${DOCKER_TAG}"
+
+# Executes a command and retries if it fails.
+# NOTE: This function is the exact copy from init-docker.sh.
+# Reason for not invoking init.docker.sh directly is since that script 
+# also performs cleanup, which we do not want in this case.
+execute() {
```

Comment:
My understanding is that we want to get rid of building docker image on the fly as part of the build step. See issue : #1719 . It will be easier to remove this step and consume the prebuilt images from dockerhub in the build step. Once issue 1719 is resolved, we can simply remove this script.